bindings/java/src/org/sleuthkit/datamodel/Report.java

```diff
@@ -90,4 +115,186 @@ public String getSourceModuleName() {
 	public String getReportName() {
 		return reportName;
 	}
+
+	@Override
+	public int read(byte[] buf, long offset, long len) throws TskCoreException {
+		if (Files.isDirectory(path)) {
+			return 0;
+		}
+
+		int totalBytesRead = 0;
+		ByteBuffer data = ByteBuffer.wrap(buf);
+		try {
+			fileChannel = FileChannel.open(path, READ);
+			fileChannel.position(offset);
+			int bytesRead = 0;
+			do {
+				bytesRead = fileChannel.read(data);
+				if (bytesRead != -1) {
+					totalBytesRead += bytesRead;
+				}
+			} while (bytesRead != -1 && data.hasRemaining());
+		} catch (IOException ex) {
+			LOGGER.log(Level.SEVERE, "Failed to read report file.", ex);
+		}
+
+		return totalBytesRead;
+	}
+
+	@Override
+	public void close() {
+		try {
+			fileChannel.close();
+		} catch (IOException ex) {
+			LOGGER.log(Level.WARNING, "Failed to close report file.", ex);
+		}
+	}
+
+	@Override
+	public long getSize() {
+		try {
+			return Files.size(path);
+		} catch (IOException ex) {
+			LOGGER.log(Level.SEVERE, "Failed to get size of report.", ex);
+			// If we cannot determine the size of the report, return zero
+			// to prevent attempts to read content.
+			return 0;
+		}
+	}
+
+	@Override
+	public <T> T accept(ContentVisitor<T> v) {
+		throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
```

Comment:
What does it mean that we support SleuthkitItemVistitor but not ContentVisitor? Is this just an interim implementation detail?

Reply:
Yes, ContentVisitor support is coming along in the "search" related code changes.

Reply:
Okay, thanks, just checking.